luarules/gadgets/unit_prevent_self_guard.lua

```diff
@@ -14,17 +14,15 @@ if not gadgetHandler:IsSyncedCode() then
     return
 end
 
+local isFactory = {} -- Factory order queue goes to their units, not themselves.
+for unitDefID, unitDef in ipairs(UnitDefs) do
+	isFactory[unitDefID] = unitDef.isFactory and #unitDef.buildOptions > 0
```

Comment:
would probably be slightly better without the buildoptions check (a zero-buildoptions factory still doesn't act on the queue and can have buildoptions added dynamically, e.g. in a mod or scenario)